### PR TITLE
Allow setting max locks, make default much higher than normal pg

### LINKED ISSUE #5157: Review 'max_locks_per_transaction' setting in FlowDB
Postgres stores table locks in memory, with pre-allocated space for `max_locks_per_transaction * max_connections` locks. Attempting to take more locks than this at any one time will result in an out-of-memory error. In the FlowDB defaults we set `max_connections=200`, and we're not explicitly setting `max_locks_per_transaction` so that defaults to 64.

Queries that select from partitioned tables require an `AccessShareLock` on each partition, as well as on every index on each of those partitions (this is true even if the query only requires data from one partition - postgres needs to obtain a lock before checking constraints to assess whether a partition is required). The events tables have a partition per day, and typically a number of indexes on each partition, so this can result in a large number of locks required for each query.

E.g. if the `events.calls` table contains 2 years of data, with 3 indexes (on `datetime`, `msisdn` and `location_id`),  then an `EventTableSubset` query such as:
```
SELECT datetime, location_id, msisdn
FROM events.calls 
WHERE events.calls.datetime >= '2022-02-10 00:00:00' AND events.calls.datetime < '2022-02-11 00:00:00'
```
would require `365*2*(3+1)=2920` 'AccessShareLock' table locks. With the current default settings, just 5 such transactions (of a possible 200) would be sufficient to exhaust the memory allocated for `200*64=12800` locks.

From this I think it's clear that the default `max_locks_per_transaction=64` is insufficient for our heavy use of partitions, and we need to increase it. A suitable value would depend on the maximum span of data we expect to keep in the DB at any given time, along with the typical number of indexes we create on the events tables.

At the same time it's worth considering whether this is the right solution. As far as I can tell there's no major reason it would be a bad idea to allow significantly more table locks per transaction, but it's possible this issue is a sign we should be approaching something a bit differently.

### PR DESCRIPTION
**Pull request by greenape**
Closes #5157

### I have:

- [ ] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [ ] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

Adds a new env var that can control Postgres' max_locks_per_transaction setting to the configure script, and increases the default value.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202306260479520

### CODE REVIEW

File: flowdb/bin/build/configurate.py [hunk 1]
```diff
@@ -69,6 +69,8 @@ def bool_env(var):
 stats_target = int(
     os.getenv("STATS_TARGET", 10000)
 )  # Default to higher than pg default
+max_locks = int(os.getenv("MAX_LOCKS", 365 * 5 * 4 * (1 + 4)))
```
Comment (Thingus) [inline flowdb/bin/build/configurate.py:72]: For future reference, what's the logic for the default here?

Comment (jc-harrison) [reply flowdb/bin/build/configurate.py:72]: https://github.com/Flowminder/FlowKit/issues/5157#issuecomment-1131735184

**Review by Thingus — APPROVED** [review]
:+1:

**Review by jc-harrison — COMMENTED** [review]
A couple of things:

I'm not sure where's best to put this info (perhaps in the secrets quickstart; probably also somewhere in the docs), but I think it's important to note (given the non-negligible memory requirement for this many locks) that when running flowdb in a docker container, the size of the /dev/shm tmpfs mount needs to be larger than
```
SHARED_BUFFERS_SIZE + MAX_LOCKS * (max_connections + max_prepared_transactions) * (sizeof(LOCK) + sizeof(LOCKTAG))
```
(otherwise postgres may attempt to allocate more shared memory than is available).

Also, I think I'd prefer to call the env variable `MAX_LOCKS_PER_TRANSACTION` rather than `MAX_LOCKS` - then it's consistent with the name of the postgres config parameter, just to avoid any ambiguity about its purpose.

**Review by jc-harrison — APPROVED** [review]
👍

**Pull request merged**